src/reactivity/ref.ts

```diff
@@ -86,12 +88,17 @@ interface RefOption<T> {
   set?(x: T): void;
 }
 class RefImpl<T> implements Ref<T> {
+  readonly [_refBrand]!: true;
   public value!: T;
   constructor({ get, set }: RefOption<T>) {
     proxy(this, 'value', {
       get,
       set,
     });
+    Object.defineProperty(this, _refBrand, {
```

Comment:
I believe we don't need this as the problem is only on type level. i.e. We only need to add the brand property on type level.